src/main/scala/singleton/ops/impl/GeneralMacros.scala

```diff
@@ -1244,6 +1279,33 @@ trait GeneralMacros {
       case CalcVal(at : String, att) => CalcVal(at.length, q"$att.length")
       case _ => unsupported()
     }
+    def RegexMatches : Calc = (a, b) match {
+      case (CalcLit.String(at), CalcLit.String(bt)) => CalcLit(at.r.matches(bt))
+      case (CalcVal(at : String, att), CalcVal(bt : String, btt)) => CalcNLit(Primitive.Boolean, q"$att.r.matches($btt)")
+      case _ => unsupported()
+    }
+    def RegexFirst : Calc = (a, b) match {
+      case (CalcLit.String(at), CalcLit.String(bt)) => CalcLit(at.r.findFirstIn(bt).get)
```

Comment:
In case of literal handling where the operation might generate an exception, we should generate an equivalent compile-time error. So either use `try catch` and send the message as a compile-time error via `abort(...)` or use a match and create your own custom error message (again, via `abort`). `abort` makes sure that the error will be directed a proper `@implicitNotFound` annotation.